src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -419,17 +464,18 @@ public override FormulaValue Visit(ErrorNode node, SymbolContext context)
             });
         }
 
-        public override FormulaValue Visit(ColorLiteralNode node, SymbolContext context)
+        public override async ValueTask<FormulaValue> Visit(ColorLiteralNode node, SymbolContext context)
         {
             return CommonErrors.NotYetImplementedError(node.IRContext, "Color literal");
         }
 
-        public override FormulaValue Visit(ChainingNode node, SymbolContext context)
+        public override async ValueTask<FormulaValue> Visit(ChainingNode node, SymbolContext context)
         {
+            CheckCancel();
```

Comment:
Referring to my previous comment with sporadic appearance of CheckCancel - why do we check here and not in the method above? If I'm a new dev adding a new method, how do I decide to call CheckCancel or not?